src/full/Agda/Syntax/Concrete/Definitions.hs

```diff
@@ -196,16 +195,15 @@ data DeclarationException'
 
 
 data DeclarationWarning = DeclarationWarning
-  { dwFileLine :: AgdaSourceErrorLocation
+  { dwLocation :: CallStack
   , dwWarning  :: DeclarationWarning'
-  } deriving (Show, Data)
+  } deriving (Show)
```

Comment:
A note here: the `Data` instance was unused. GHC's `CallStack` does not have an instance of `Data CallStack` by default, though it can be derived as an orphan if it were ever needed.

Reply:
We need to check but I think we introduced most of the `Data` constraints
to be able to [check equality of constructors for warnings](https://stackoverflow.com/questions/43944044/checking-that-two-values-have-the-same-head-constructor), something we
do not need anymore AFAIK.

Going through the modules and trying to remove as many `Data` deriving
as possible has been in the back of my mind for a while but I never really
got around to it. I wish there was an easy way to see a dependency graph
of these instances and their uses... :(

Reply:
> I wish there was an easy way to see a dependency graph of these instances and their uses... :(

I definitely agree. Tracking instance dependencies is far more tricky than it should be. And the obvious tactic of removing instances and seeing what breaks is (sometimes, potentially, in general) unsafe due to overlapping constraints and so on.